src/sidebar/components/Annotation/AnnotationEditor.tsx

```diff
@@ -137,6 +144,29 @@ function AnnotationEditor({
     [annotation, draft, isReplyAnno, store],
   );
 
+  const defaultAuthority = store.defaultAuthority();
+  const mentionMode = useMemo(
+    (): MentionMode =>
+      isThirdPartyUser(annotation.user, defaultAuthority)
+        ? 'display-name'
+        : 'username',
+    [annotation.user, defaultAuthority],
+  );
+  const usersDisplayNamesMapRef = useRef<Map<string, UserItem>>(
+    new Map(
+      mentionMode === 'username'
+        ? []
+        : // If the annotation is being edited, it may have mentions. Use them to
+          // initialize the display names map
+          annotation.mentions
+            ?.filter(mention => !!mention.display_name)
+            .map(({ userid, username, display_name: displayName }) => [
+              displayName!,
+              { userid, username, displayName },
+            ]),
+    ),
+  );
```

Comment:
I'm not 100% sure if a ref is the right way to track the map of display names. If the annotation changed while the editor is open, it could cause the old mentions to still be incorrectly tracked, but as far as I can tell, that's something that cannot happen, as every annotation has its own editor component instance.

I didn't use a memoized value because the map starts with an initial list of students, but then it needs to be editable.

I didn't use a regular piece of state (`useState`) because the component doesn't need to re-render when the map changes. Also, trying to address the issue I explained above with a piece of state would require updating it in a side effect, which is tricky to do right.

Reply:
Actually, I'm thinking a possible solution would be to not initialize this here based on `annotation.mentions`. Instead, use it to track inserted suggestions only, and at the moment of saving the annotation, then "merging" that with `annotation.mentions`. That way the map will match the annotation at the moment of saving.